### PR TITLE
Do not send a symbol search request to OmniSharp if search term is shorter than "omnisharp.minFindSymbolsFilterLength" 

### PR DESCRIPTION
**Pull request by filipw**
Instead of doing the validation on the server, we do it on the client.

### CODE REVIEW

File: src/features/workspaceSymbolProvider.ts [hunk 1]
```diff
@@ -23,10 +23,15 @@ export default class OmnisharpWorkspaceSymbolProvider extends AbstractSupport im
         let options = this.optionProvider.GetLatestOptions();
         let minFilterLength = options.minFindSymbolsFilterLength > 0 ? options.minFindSymbolsFilterLength : undefined;
```
Comment (dmgonch) [inline src/features/workspaceSymbolProvider.ts]: >minFindSymbolsFilterLength [](start = 38, length = 26)

Sounds like you could remove the setting as well.

Comment (filipw) [reply src/features/workspaceSymbolProvider.ts]: the setting makes sense, it's just we evaluate it on the client rather than server

Comment (dmgonch) [reply src/features/workspaceSymbolProvider.ts]: you are right - sorry for the confusion.

File: src/features/workspaceSymbolProvider.ts [hunk 2]
```diff
@@ -23,10 +23,15 @@ export default class OmnisharpWorkspaceSymbolProvider extends AbstractSupport im
         let options = this.optionProvider.GetLatestOptions();
         let minFilterLength = options.minFindSymbolsFilterLength > 0 ? options.minFindSymbolsFilterLength : undefined;
         let maxItemsToReturn = options.maxFindSymbolsItems > 0 ? options.maxFindSymbolsItems : undefined;
-        return serverUtils.findSymbols(this._server, { Filter: search, MinFilterLength: minFilterLength, MaxItemsToReturn: maxItemsToReturn, FileName: '' }, token).then(res => {
```
Comment (dmgonch) [inline src/features/workspaceSymbolProvider.ts]: But FindSymbolsRequest.MinFilterLength can be removed right, along with removing it on the server side, right?

Comment (filipw) [reply src/features/workspaceSymbolProvider.ts]: yes, I removed it from the protocol.ts, thanks for the pointer 👍

**Comment by dmgonch** [comment]
@filipw : let me know if you working on removing MinFilterLength from OmniSharp/omnisharp-roslyn as well - I will pick up the cleanup otherwise. thanks.

**Comment by filipw** [comment]
we will not remove it yet to not cause a breaking change on the API, since it already shipped. I will remove it in the future, but not yet

**Pull request merged**